eth/gasprice/gasprice.go

```diff
@@ -152,26 +152,32 @@ type getBlockPricesResult struct {
 	err   error
 }
 
-// getLowestPrice calculates the lowest transaction gas price in a given block
+type transactionsByGasPrice []*types.Transaction
+
+func (t transactionsByGasPrice) Len() int { return len(t) }
+func (t transactionsByGasPrice) Swap(i, j int) { t[i], t[j] = t[j], t[i] }
+func (t transactionsByGasPrice) Less(i, j int) bool { return t[i].GasPrice().Cmp(t[j].GasPrice()) < 0 }
+
+// getBlockPrices calculates the lowest transaction gas price in a given block
 // and sends it to the result channel. If the block is empty, price is nil.
 func (gpo *Oracle) getBlockPrices(ctx context.Context, signer types.Signer, blockNum uint64, ch chan getBlockPricesResult) {
 	block, err := gpo.backend.BlockByNumber(ctx, rpc.BlockNumber(blockNum))
 	if block == nil {
 		ch <- getBlockPricesResult{nil, err}
 		return
 	}
+
 	txs := block.Transactions()
-	var minPrice *big.Int
+	sort.Sort(transactionsByGasPrice(txs))
```

Comment:
`block.Transactions()` returns the internal slice (https://github.com/ethereum/go-ethereum/blob/master/core/types/block.go#L287), which means sorting it will have nasty side effects.

Reply:
Uhh, good catch!